conformance/lang/expressions/object-constructor-expr/object_remote_methods.balt

```diff
@@ -0,0 +1,407 @@
+    io:println(res); // @output 7
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present,
+             where non-isolated module-level state cannot be mutated.
+Labels: additive-expr, array-type, field-access-expr, int, intersection-type, list-constructor-expr, member-access-expr,
+        method-call-expr, remote-method-defn, module-init-var-decl, object-constructor-expr, object-field, readonly-type, var
+
+int[] & readonly arr = [3, 4];
+
+function errorFunction() {
+    var _ = client object {
+        remote isolated function func1() returns int {
+            return arr[1]; // @error invalid access of mutable storage in an isolated function
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present,
+             where isolated module-level state is read in lock statement.
+Labels: additive-expr, array-type, client-remote-method-call-action, int, isolated-function, list-constructor-expr,
+        lock-stmt, member-access-expr, method-call-expr, remote-method-defn, module-init-var-decl, object-constructor-expr,
+        object-field, var
+
+isolated int[] arr = [3, 4];
+
+function init() {
+    var obj = client object {
+        remote isolated function func1() returns int {
+            lock {
+                arr[2] = 5;
+                return arr[0] + arr[2];
+            }
+        }
+    };
+
+    int res = obj->func1();
+    io:println(res); // @output 8
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present,
+             where non-isolated module-level state cannot be mutated.
+Labels: int, isolated-function, remote-method-defn, module-init-var-decl, object-constructor-expr, object-field, var
+
+int intVal = 10;
+
+function errorFunction() {
+    var _ = client object {
+        remote isolated function func1() {
+            lock {
+                intVal = 5; // @error invalid access of mutable storage in an isolated function
+            }
+        }
+    };
+}
+
+Test-Case: output
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present,
+             where captured variables must be both final and have a static type that is a subtype of
+             readonly|isolated object {}.
+Labels: client-remote-method-call-action, explicit-anonymous-function-expr, int, method-call-expr, remote-method-defn,
+        object-constructor-expr, object-field, var
+
+function init() {
+    final int val = 10;
+    var obj = client object {
+        remote isolated function func1() returns int => val;
+    };
+
+    int res = obj->func1();
+    io:println(res); // @output 10
+}
+
+Test-Case: error
+Description: Test method-defn-body satisfying the requirements for an isolated function if isolated-qual is present,
```

Comment:
Shall we rephrase the descriptions just to include the tested scenario? Later test as well